src/manager/locked_load.rs

```diff
@@ -0,0 +1,71 @@
+use uuid::Uuid;
+
+use crate::store::EventStoreLockGuard;
+use crate::AggregateState;
+
+/// The outcome of [`crate::manager::AggregateManager::lock_and_load`], akin to `Option<AggregateState<_>>`.
+///
+/// it contains the loaded [`AggregateState`] if found,
+/// and ensures that the lock is preserved in all cases.
+///
+/// Releases the lock on drop, unless it gets transferred through its API.
+pub struct LockedLoad<T>(LockedLoadInner<T>);
+
+impl<T> LockedLoad<T> {
+    /// Constructs a new instance from an AggregateState.
+    /// It should contain its own lock.
+    pub fn some(aggregate_state: AggregateState<T>) -> Self {
+        Self(LockedLoadInner::Some(aggregate_state))
+    }
+
+    /// Constructs a new instance to hold the lock for an empty state.
+    pub fn none(id: Uuid, lock: EventStoreLockGuard) -> Self {
+        Self(LockedLoadInner::None { id, lock })
+    }
+
+    /// Checks if the AggregateState was found.
+    pub fn is_some(&self) -> bool {
+        let Self(inner) = self;
+        match inner {
+            LockedLoadInner::None { .. } => false,
+            LockedLoadInner::Some(_) => true,
+        }
```

Comment:
What about something like?
```suggestion
        matches!(self, Self(LockedLoadInner::Some(_)))
```